docs/source/security/cors.mdx

```diff
@@ -0,0 +1,123 @@
+
+Below is an example of setting up CORS with the [`apollo-server-express`](../integrations/middleware#swapping-out-apollo-server) package:
+
+```js
+// ... set up the server
+await server.start();
+
+const corsOptions = {
+  origin: ["http://localhost:3000", "https://studio.apollographql.com"]
+};
+
+server.applyMiddleware({
+  app,
+  cors: corsOptions,
+  path: "/graphql",
+});
+```
+
+### Configuring origins
+
+CORS uses [specific HTTP request headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#the_http_response_headers) as part of its protocol, including[`Access-Control-Allow-Origin`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin). The `Access-Control-Allow-Origin` header enables a server to dictate which origins can use scripts to access that server's resources. Using [`cors`](https://github.com/expressjs/cors#configuration-options) middleware, you can configure the `Access-Control-Allow-Origin` header using the `origins` option.
+
+By default, the batteries-included `apollo-server` package uses the wildcard value (`*`) to allow requests from any origin _without credentials_ to access its served responses. This works great in development, allowing you to delay configuring CORS until your application is ready for production. You can also [remove CORS middleware entirely](../api/apollo-server#cors), disabling cross-origin requests.
+
+> If you need to pass credentials to your server (e.g., via cookies or an HTTP header), you can't use the wildcard value (`*`) for your origin. You _must_ provide specific origins; see [Passing credentials with CORS](#passing-credentials-with-cors) for examples.
+
+When your server is in production, we strongly recommend listing the specific origins that should have access to your server:
+
+```js
+//  With the batteries-included Apollo Server
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+  cors: {
+    origin: ["https://www.exampleApp.com", "https://studio.apollographql.com"]
+  },
+});
+```
+
+> Note if you plan to use [Apollo Studio Explorer](https://www.apollographql.com/docs/studio/explorer/explorer/) as a GraphQL web IDE you should include `https://studio.apollographql.com` in your list of valid origins.
+
+In certain situations, you may want to use the wildcard value (`*`) in production, allowing requests from any origin. For example, if you create a public API or an API to embed in other websites, you might want to allow all origins to access your server's resources. If you do use the wildcard value in production, _any_ website can tell a user's browser to send an arbitrary request (without cookies or other credentials) to your server and read that server's response.
+
+## Passing credentials with CORS
+
+If your cross-site requests [include a user's credentials](https://developer.mozilla.org/en-US/docs/Web/API/Request/credentials) (e.g., cookies or HTTP Authentication headers), you might need to modify your CORS configuration.
```

Comment:
If capitalizing Authentication, use `codefont` to indicate the canonical header name